### PR TITLE
[Bug] Fixes Unseen Fist ignoring abilities

### PR DESCRIPTION
**Pull request by schmidtc1**
## What are the changes the user will see?

Unseen Fist + direct contact moves allow the user to ignore abilities (e.g. Sturdy)

## Why am I making these changes?

It's a bug
Discord report: https://discord.com/channels/1125469663833370665/1274675083872436235

## What are the changes from a developer perspective?

Adds a break in the switch case, so that it doesn't fall through to `IGNORE_PROTECT` when checking if a move should ignore abilities.

### Screenshots/Videos

https://github.com/user-attachments/assets/3307af07-dd60-4cf6-accf-c57dd0a7c26b

## How to test the changes?

Give pokemon Unseen Fist + direct contact move. Give opponent Sturdy, see that it no longer KOs through Sturdy

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [n/a] If I have text, did I add placeholders for them in locales?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by innerthunder — DISMISSED** [review]

File: src/data/move.ts [hunk 1]
```diff
@@ -595,6 +595,7 @@ export default class Move implements Localizable {
           return true;
         }
       }
+      break;
     case MoveFlags.IGNORE_PROTECT:
```
Comment (torranx) [inline src/data/move.ts:599]: shouldn't this have `break` too?

Comment (innerthunder) [reply src/data/move.ts:599]: It's the last case in the switch block, so it doesn't *really* matter, but it'd be nice to have if people (like me) implement new cases and forget to add the break later.

Comment (schmidtc1) [reply src/data/move.ts:599]: Yea, I'll add it to be safe

**Comment by schmidtc1** [comment]
Accidentally pushed my overrides in https://github.com/schmidtc1/pokerogue/commit/2589faf69edd67883272761ab3ddf59a54e8c013. 
Reverted and should be fixed

**Review by torranx — APPROVED** [review]

**Review by innerthunder — APPROVED** [review]

**Review by frutescens — COMMENTED** [review]
Tested and works. Does this change also apply to Pokemon with Fluffy? I tried testing it but it was hard to tell. 

https://github.com/user-attachments/assets/79f7d038-fce8-49a6-afe8-8524ff5900ef

**Comment by schmidtc1** [comment]
> Tested and works. Does this change also apply to Pokemon with Fluffy? I tried testing it but it was hard to tell.
> 
>  Screen.Recording.2024-08-27.at.11.56.05.AM.mov

Yes this also applies to Fluffy. Essentially Unseen Fist was acting as if it were Mold Breaker.

**Review by frutescens — APPROVED** [review]

**Pull request merged**